core/adapters/http_test.go

```diff
@@ -110,10 +110,8 @@ func TestHTTPGet_Perform(t *testing.T) {
 }
 
 func TestHTTPGet_TimeoutAllowsRetries(t *testing.T) {
-	t.Parallel()
```

Comment:
Does this alone do the trick or do you also need 30->80ms below?

Reply:
It doesn't fix it locally with `GOMAXPROCS=1`, the timing change was also needed. It won't run any tests in parallel but it can still take time to do a `GC` sweep or context switch to execute the test runner internal goroutine.